### PR TITLE
add modinv

### LINKED ISSUE #2: Add `modinv` operator
To fully prove _all_ circom circuits we need a modular inverse operator (similar to division). The operator should work as

`inv 0x2 0x0` - Calculate the modular inverse of value in register `0x0` and store the result in `0x2`. It should be possible for output and input register to be the same (e.g. modinv in place).

A constraint for this should look like `opcode_selector * (out * r1 - 1)`. `out` is the output register and `r1` is the input register.

A free input should be used to store an intermediate value to make this constraint <= degree 3.

### PR DESCRIPTION
**Pull request by joyqvq**
Closes #2

### CODE REVIEW

File: test/operators.test.mjs [hunk 1]
```diff
@@ -118,6 +118,7 @@ eq 0x2 0x1
 
 test('should negate in place', async t => {
   const v = field.random()
+  console.log('hi', v.toString())
```
Comment (vimwitch) [inline test/operators.test.mjs:121]: This should be removed

File: test/operators.test.mjs [hunk 2]
```diff
@@ -158,3 +159,16 @@ eq 0x0 0x1
   await proveAndVerify(asm)
   t.pass()
 })
+
+test('should mod inverse in place', async t => {
```
Comment (vimwitch) [inline test/operators.test.mjs:163]: Could you add another test but storing the output in a different register?

**Review by vimwitch — COMMENTED** [review]
Thank you, PR looks good, just a couple small changes

File: test/operators.test.mjs (hunk 2 above)
Comment (joyqvq) [reply test/operators.test.mjs]: you mean, output to 0x0 instead of 0x2? added

**Pull request merged**